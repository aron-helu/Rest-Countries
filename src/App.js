import { Route, Routes } from 'react-router-dom';
import Countries from './Components/Countries/Countries';
import Home from './Components/Home/Home';
import NavBar from './Components/NavBar/NavBar';

const App = () => (
  <div className="App">
    <NavBar />
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/:name" element={<Countries />} />
    </Routes>
  </div>
);

export default App;
