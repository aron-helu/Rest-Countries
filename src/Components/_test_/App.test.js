import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../../redux/configureStore';
import App from '../../App';

test('snapshot for App', () => {
  const tree = render(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  );
  expect(tree).toMatchSnapshot();
});
