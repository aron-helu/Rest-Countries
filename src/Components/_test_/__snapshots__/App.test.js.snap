// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <nav
          class="navigation-bar"
        >
          <ul
            class="navigation-ul"
          >
            <li
              class="NavLink"
            >
              <a
                aria-current="page"
                class="log active"
                href="/"
              >
                COUNTRY
              </a>
            </li>
          </ul>
          <div
            class="logo-div"
          >
            <img
              alt="world"
              class="ar-img"
              src="aron-transparent.png"
            />
            <p
              class="logo-p"
            >
              Where in the world?
            </p>
          </div>
        </nav>
        <div
          class="characters-section"
        >
          <div
            class="ar-characters-title"
          >
            <div
              class="ar-characters-text"
            >
              <p
                class="text"
              >
                 Continent 
              </p>
            </div>
          </div>
          <div
            class="ar-characters"
          >
            <div
              class="character-title"
            >
              Filter By Region
            </div>
            <div
              class="fix"
            >
              <input
                class="input"
                placeholder="Search - continent"
                type="text"
              />
              <div
                class="vitor"
              >
                <div
                  class="individual-character"
                >
                  <a
                    href="/Africa"
                  >
                    <h2
                      class="continent-title"
                    >
                      Africa
                    </h2>
                  </a>
                </div>
                <div
                  class="individual-character"
                >
                  <a
                    href="/Asia"
                  >
                    <h2
                      class="continent-title"
                    >
                      Asia
                    </h2>
                  </a>
                </div>
                <div
                  class="individual-character"
                >
                  <a
                    href="/Oceania"
                  >
                    <h2
                      class="continent-title"
                    >
                      Oceania
                    </h2>
                  </a>
                </div>
                <div
                  class="individual-character"
                >
                  <a
                    href="/Europe"
                  >
                    <h2
                      class="continent-title"
                    >
                      Europe
                    </h2>
                  </a>
                </div>
                <div
                  class="individual-character"
                >
                  <a
                    href="/North America"
                  >
                    <h2
                      class="continent-title"
                    >
                      North America
                    </h2>
                  </a>
                </div>
                <div
                  class="individual-character"
                >
                  <a
                    href="/South America"
                  >
                    <h2
                      class="continent-title"
                    >
                      South America
                    </h2>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <nav
        class="navigation-bar"
      >
        <ul
          class="navigation-ul"
        >
          <li
            class="NavLink"
          >
            <a
              aria-current="page"
              class="log active"
              href="/"
            >
              COUNTRY
            </a>
          </li>
        </ul>
        <div
          class="logo-div"
        >
          <img
            alt="world"
            class="ar-img"
            src="aron-transparent.png"
          />
          <p
            class="logo-p"
          >
            Where in the world?
          </p>
        </div>
      </nav>
      <div
        class="characters-section"
      >
        <div
          class="ar-characters-title"
        >
          <div
            class="ar-characters-text"
          >
            <p
              class="text"
            >
               Continent 
            </p>
          </div>
        </div>
        <div
          class="ar-characters"
        >
          <div
            class="character-title"
          >
            Filter By Region
          </div>
          <div
            class="fix"
          >
            <input
              class="input"
              placeholder="Search - continent"
              type="text"
            />
            <div
              class="vitor"
            >
              <div
                class="individual-character"
              >
                <a
                  href="/Africa"
                >
                  <h2
                    class="continent-title"
                  >
                    Africa
                  </h2>
                </a>
              </div>
              <div
                class="individual-character"
              >
                <a
                  href="/Asia"
                >
                  <h2
                    class="continent-title"
                  >
                    Asia
                  </h2>
                </a>
              </div>
              <div
                class="individual-character"
              >
                <a
                  href="/Oceania"
                >
                  <h2
                    class="continent-title"
                  >
                    Oceania
                  </h2>
                </a>
              </div>
              <div
                class="individual-character"
              >
                <a
                  href="/Europe"
                >
                  <h2
                    class="continent-title"
                  >
                    Europe
                  </h2>
                </a>
              </div>
              <div
                class="individual-character"
              >
                <a
                  href="/North America"
                >
                  <h2
                    class="continent-title"
                  >
                    North America
                  </h2>
                </a>
              </div>
              <div
                class="individual-character"
              >
                <a
                  href="/South America"
                >
                  <h2
                    class="continent-title"
                  >
                    South America
                  </h2>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
